copy/single.go

```diff
@@ -773,12 +773,12 @@ func (ic *imageCopier) copyLayer(ctx context.Context, srcInfo types.BlobInfo, to
 				bar.mark100PercentComplete()
 				hideProgressBar = false
 				logrus.Debugf("Retrieved partial blob %v", srcInfo.Digest)
-				return true, updatedBlobInfoFromUpload(srcInfo, uploadedBlob)
+				return true, updatedBlobInfoFromUpload(srcInfo, uploadedBlob), nil
 			}
 			logrus.Debugf("Failed to retrieve partial blob: %v", err)
-			return false, types.BlobInfo{}
-		}(); reused {
-			return blobInfo, cachedDiffID, nil
+			return false, types.BlobInfo{}, err
```

Comment:
I can see how we might need this for the ComposeFS-only operation, but right now (and still as of https://github.com/containers/storage/pull/1833 ), if we have a chunked layer but c/storage chunked pulling is disabled, AFAICS this would cause an irrecoverable error instead of falling back to the non-chunked layer creation path.

(The authoritative documentation for `PutBlobPartial`, right now, explicitly says that the caller should fall back on any error.)

So it seems to me that this would require a bit more infrastructure to be built. Can this part be split into a separate PR?

Reply:
This is the cause of the Skopeo test failures .

Reply:
yes let's not rush this change, I've dropped it